modules/light-clients/08-wasm/types/contract_api.go

```diff
@@ -7,8 +7,9 @@ import (
 
 // InstantiateMessage is the message that is sent to the contract's instantiate entry point.
 type InstantiateMessage struct {
-	ClientState    *ClientState    `json:"client_state"`
-	ConsensusState *ConsensusState `json:"consensus_state"`
+	ClientState    []byte `json:"client_state"`
+	ConsensusState []byte `json:"consensus_state"`
+	Checksum       []byte `json:"checksum"`
```

Comment:
I left the checksum here for now.

Reply:
What about latest height. Also, I don't see why contracts would need their own checksum. I think we should remove this.

Reply:
I didn't add it because the contract takes the [latest height from the underlying client state (the bytes that we are passing here) and uses that to set the latest height of the 08-wasm wrapping type](https://github.com/misko9/centauri/blob/master/light-clients/ics10-grandpa-cw/src/client.rs#L306).

Happy to remove the checksum if you do me the favour to figure out how the contract can calculate that itself. I don't think I will have the time to investigate that. 🙏

Reply:
Happy to keep. Contracts can only get this through a querier which we are passing `nil`. I think this was the only way to get it for standard contracts. I still don't see the reason why the contract should be aware of its own codehash.

Also it can get its own code hash during other executions from the clientStore right? Eitherway. Happy to keep.

Reply:
I think we should assume that the contract cannot get it's own codehash in instantiate if we remove this since we are not passing in a querier. But it can get it during other executions. I think the contract does not need it's own codeHash in instantiate and we could remove this. I'm in favour of removal but happy to hear what others think